standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -431,7 +432,8 @@ public void setThreadFactory(ThreadFactory threadFactory) {
     processor = new ThriftHiveMetastore.Processor<>(handler);
     LOG.info("Starting DB backed MetaStore Server with generic processor");
     boolean jwt = MetastoreConf.getVar(conf, ConfVars.THRIFT_METASTORE_AUTHENTICATION).equalsIgnoreCase("jwt");
-    ServletSecurity security = new ServletSecurity(conf, jwt);
+    AuthType authType = jwt ? AuthType.JWT : AuthType.SIMPLE;
```

Comment:
why do we need these hacks? can we drop the jwt var completely?

I think we should replace `ICEBERG_CATALOG_SERVLET_AUTH` & `PROPERTIES_SERVLET_AUTH`
with
`METASTOTE_REST_CATALOG_AUTHENTICATION("metastore.rest-catalog.authentication"`
````
String authType = MetastoreConf.getVar(configuration, ConfVars.METASTOTE_REST_CATALOG_AUTHENTICATION);
    ServletSecurity security = new ServletSecurity(AuthType.fromString(authType), configuration);
`

Reply:
This line configures [an HTTP server for HiveMetaStore with Thrift over HTTP](https://github.com/apache/hive/blob/174ff56b77b212bf51ee01587de9cf22e77f2dd3/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java#L966-L977), not for Iceberg REST or Properties REST. This change keeps the original behavior. We can't fully encode `hive.metastore.authentication` there, as it doesn't have SIMPLE.

As for the necessity of consolidation, I am currently biased toward separation. Let's say we introduced OAuth 2.0 with OIDC for the Iceberg REST catalog, and a user wanted to use it. Can all PROPERTIES clients immediately use OAuth? I am a little skeptical that different HTTP endpoints can share the same authentication method.

Reply:
ok, how about 
````
String authType = MetastoreConf.getVar(configuration, ConfVars.THRIFT_METASTORE_AUTHENTICATION);

ServletSecurity security = new ServletSecurity(
  AuthType.JWT.name().equalsIgnoreCase(authType) ? AuthType.JWT : AuthType.SIMPLE, 
  configuration);
````